### ISSUE TITLE
Interesting artefact with polygon.buffer

### ISSUE THREAD

**Issue by eivindlm** [Toblerity/Shapely#1044]
## Expected behavior and actual behavior.

I use the buffer method on a polygon object (single polygon). I expect to get a single polygon back, but a MultiPolygon is returned with some artefact objects along with the buffer result. I will attach a file to the issue with the data that triggers the behaviour.

The issue occurs only for buffer values in the range [14608, 16656]. For values below or above this range, a Polygon is returned.

## Steps to reproduce the problem.

from shapely.geometry import Polygon
from poly_coords import poly_coords

polygon=Polygon(poly_coords)
print(type(polygon))   # Prints "Polygon"
print(polygon.is_valid)  # Prints "True"

buffered_polygon=polygon.buffer(15000)
print(type(buffered_polygon))  # Prints "MultiPolygon"  ==> expected "Polygon"
print(len(buffered_polygon))   # Prints 3  ==> expected 1

## Operating system

Linux, Fedora 32

## Shapely version and provenance

1.7.1 installed from PyPi using pip.
python 3.7

**Comment by eivindlm** [comment]
Data to reproduce the problem.

[poly_coords.zip](https://github.com/Toblerity/Shapely/files/5666857/poly_coords.zip)

**Comment by mwtoews** [comment]
Unfortunately, this is an upstream bug with GEOS that we can't fix in Shapely. This is also a bug with JTS and GEOS master (i.e. still an issue with OverlayNG -- perhaps @dr-jts is interested by these cases).

I've seen these peculiar issues before, where short segments create distortions with the buffered output. There are two workarounds:
1. Simplify the input to remove small segments that are likely the cause of the buffer issue:
```python
simple_polygon = polygon.simplify(0.0001)
print(polygon.area - simple_polygon.area)  # -0.006700038909912109
simple_buffered_polygon = simple_polygon.buffer(15000)
print(simple_buffered_polygon.geom_type)  # Polygon
print(buffered_polygon.area - simple_buffered_polygon.area)  # -60.210787773132324
```
2. If you are only expecting one part, take the largest part:
```python
biggest_buffered_polygon = sorted(buffered_polygon.geoms, key=lambda g: g.area)[-1]
print(buffered_polygon.area - biggest_buffered_polygon.area)  # 47.43143653869629
```

**Comment by dr-jts** [comment]
Yes, I'm interested in these cases.  The buffer code has not been affected by OverlayNG. These are longstanding bugs to do with handling a very complex buffer curve.

A full fix will require substantial effort, unfortunately, so not sure when that will get done.  In the meantime, workaround #2 could be added to the buffer code as a heuristic (for the case where the input is a single geometry).
